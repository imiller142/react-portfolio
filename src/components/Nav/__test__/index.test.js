import React from "react";
import {render, cleanup, getByTestId} from '@testing-library/react';
import '@testing-library/jest-dom';
import Navigation from "..";

const categories = [
    {name: 'test', decription: 'test'}
]

const mockCurrentCategory = jest.fn();
const mockSetCurrentCategory = jest.fn();

afterEach(cleanup);

describe('Nav component', () => {
    it('render', () => {
        render(<Navigation
            categories={categories}
            setCurrentCategory={mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
            />);
    })

    it('matches snapshot', () => {
        const { asFragment } = render(<Navigation
            categories={categories}
            setCurrentCategory={mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
        />);
        expect(asFragment()).toMatchSnapshot();
    })
})

// test that rendering displays proper nav component name
describe('contains category name on nav bar', () => {
    it('render and test content', () => {
        const { getByTestId } = render(<Navigation
            categories={categories}
            setCurrentCategory={mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
        />);
        expect(getByTestId('test')).toHaveTextContent('test');
    })
})