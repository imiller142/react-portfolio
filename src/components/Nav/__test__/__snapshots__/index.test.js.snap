// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-secondary py-4 bg-img navbar navbar-expand-lg navbar-light"
  >
    <div
      class="w-25 container"
    >
      <span
        class="text-secondary navbar-brand"
      >
         Ian Miller's Portfolio
      </span>
    </div>
    <div
      class="justify-content-between container"
    >
      <div
        class="justify-content-between w-100 navbar-nav"
      >
        <div
          class="nav-item"
        >
          <a
            class="text-quad nav-link"
            data-rr-ui-event-key="#"
            data-testid="test"
            href="#"
            name="test"
            role="button"
            tabindex="0"
          >
            test
          </a>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
